python/sglang/srt/layers/quantization/fp8.py

```diff
@@ -1023,12 +1031,7 @@ def apply(
             if ret is not None:
                 return ret
 
-        if (
-            get_bool_env_var("SGLANG_CUTLASS_MOE")
-            and self.cutlass_fp8_supported
-            and self.block_quant
-            and (is_sm100_supported() or is_sm90_supported())
-        ):
+        if self.should_use_cutlass_fused_experts_fp8():
```

Comment:
1. Please cache this during the initialization of this layer
2. Can you also help cache `if use_intel_amx_backend(layer):`?

Reply:
Thanks @merrymercy I have added caching for this.

For `use_intel_max_backend`, I don't think any caching is necessary. The function is just doing a simple check on the layer: https://github.com/sgl-project/sglang/blob/f96413c444a4ce16c6b01770e28a636350df24bf/python/sglang/srt/utils.py#L223-L224